src/BSeries.jl

```diff
@@ -1453,4 +1456,379 @@ include("latexify.jl")
+    #renormalize by multiplying by symmetry factor
+    renormalize_bseries!(flow_series)
+    #save all the coefficients in an array: it is easier
+    #to call an array full of fractions when working with the 
+    #functions 'renormalize_bseries!' and 'energy_preserving_trees_test' 
+    coefficients = collect(values(flow_series))
+    #save all the RootedTrees in another array: 
+    thetrees = collect(keys(flow_series))
+    #we need only the level sequence
+    # Create an empty vector to store the converted trees into arrays
+    trees = Vector{Vector{Int}}(undef, length(flow_series))
+    # Convert the trees and store them in the 'trees' vector
+    for i in 1:length(flow_series)
+        trees[i] = thetrees[i].level_sequence
+    end
+    energy_preserving_flag = true
+    max_length = length(trees[end])
+    l = length(trees)
+    #first, check if the energy_preserving conditon is satisfied
+    #low order (provided that the computational cost at order < 5)
+    # is low. This is because the 'rank' function's behavior
+    # in the test function doesn't works by separate for low orders.
+    if low_order_energy_preserving(trees, coefficients) == false
+        return false
+    end
+    #loop over same-order trees
+    order = 5
+    while order <= max_length && energy_preserving_flag == true
+        same_order_trees = []
+        same_order_coeffs = []
+        #we create and fill the arrays of trees and coefficients
+        #corresponding to an 'order'
+        for j in 1:l
+            if length(trees[j]) == order
+                push!(same_order_coeffs, coefficients[j])
+                push!(same_order_trees,trees[j])
+            end
+        end
+        energy_preserving_flag = energy_preserving_trees_test(same_order_trees,same_order_coeffs)
+        order = order + 1
+    end
+    return energy_preserving_flag
+end
+
+#       low_order_energy_preserving(trees,coefficients)
+#
+# This function checks whether the set of level_sequences 'trees'
+# and the set of 'coefficients' corresponding to a Bseries is 
+# energy_preserving up to the 'uppermost_order', which we choose 
+# to be 4 for computational optimization
+function low_order_energy_preserving(trees, coefficients)
+    #set the limit up to which this function will work
+    uppermost_order = 4
+    same_order_trees = []
+    same_order_coeffs = []
+    index = 1
+    while length(trees[index]) <= uppermost_order
+        push!(same_order_coeffs, Rational{Int64}(coefficients[index]))
+        push!(same_order_trees,trees[index])
+        index += 1
+        if index > length(trees)
+            break
+        end
+    end
+    return energy_preserving_trees_test(same_order_trees,same_order_coeffs)
+end
+
+#        remove_trunk(a)
+
+#This function returns the forest of level_sequences 
+#obtained after removing the rightmost trunk of a given
+#level_sequence.
+function remove_trunk(a)
+    m = num_branchs(a)
+    branchs_array = Array{Array}(undef, m)
```

Comment:
It's clear that this is an array, but that doesn't help with readability.  Let's just call it `branches`.

Reply:
Also, "branches" needs to have an "e" here and everywhere that it appears.

Reply:
Implemented.

Reply:
Oh, I see that my comment wasn't clear.  I didn't mean to change the name of the function `remove_trunk`, but instead to change the name of `branches_array` to `branches`.

Reply:
Okey. I changed `branches_array` for `branches.` On the other hand, I linked the functions `get_branches` and `modify_branches` into only one function called `get_branches`.